pandas/tests/frame/test_axis_select_reindex.py

```diff
@@ -1152,3 +1152,22 @@ def test_drop_non_empty_list(self, index, drop_labels):
         # GH 21494
         with pytest.raises(KeyError, match="not found in axis"):
             pd.DataFrame(index=index).drop(drop_labels)
+
+    @pytest.mark.parametrize(
+        "operation", ["__iadd__", "__isub__", "__imul__", "__ipow__"]
+    )
+    @pytest.mark.parametrize("inplace", [False, True])
+    def test_inplace_drop_and_operation(self, operation, inplace):
+        # GH 30484
+        data_dict = {"x": [1, 2, 3, 4, 5], "y": [10, 20, 30, 40, 50]}
+        df = pd.DataFrame(data_dict)
+        y = df["y"]
+
+        if inplace:
+            df.drop("y", axis=1, inplace=inplace)
+        else:
+            df = df.drop("y", axis=1, inplace=inplace)
+        # Perform operation and ensure that df is not changed
+        expected = df.copy()
```

Comment:
can you put more in the inplace and else clauses, e.g. you should copy the expected first.

also can you assert that no SettingWithCopyWarning is show here.

Reply:
Hi @jreback,

Can you explain the first sentence please? In my opinion, since I want the test to make sure that an inplace operation in "y" doesn't change the origin "df", then the copy is done in the right place, that is right before this operation.

As for the warning, is it ok if I ensure that no warning of any type is produced?

Reply:
Hi @jreback, did you have time to read my comment? Thanks in advance

Reply:
you need to move the 

expeted = df.copy() before running the function, otherwise you can't tell if you modified the original or not.

also need to do

tm.assert_produces_warning(None):
   ......

to make sure not outputting any SettingWithCopyWarnings

Reply:
I still think that the copy is being done in the right place. It's the operation in "y" that is changing the "df" when it shouldn't. And the copy is done before this operation. If I do the copy before dropping the "y" from "df" then it's normal that I'll get a different "df" at the end in a bug-free code.

Anyway, I'll try to write the test in a different way that makes sense for you and me.

Thanks!